### PR TITLE
[bugfix] Resolve logging reduction when using sync_dist + reduce_fx={mean, max}

### LINKED ISSUE #4395: loss from progress bar appears to be sum of loss across all GPUs in Lightning 1.0.3
## 🐛 Bug

The loss from progress bar during the training appears to be sum of loss across of all GPUs

### To Reproduce

One can pick any model and dataset and varies the number of GPUs for training. In my example, I have roughly 0.7 loss for 1 GPUs, 1.4 loss for 2 GPUs and 0.28 loss for 4 GPUs **for the first few training batches**, indicating loss is roughly the **sum** instead of **mean** of loss across all GPUs.

I used the standard from the doc:

```
def train_step(self, batch, batch_idx):
    loss = self.forward(batch)
   self.log('loss/train', loss, on_step=True, on_epoch=False, sync_dist=True, sync_dist_op='mean')

```

Note: the loss logged in tensorboard appears to be correct (mean of loss)

### Expected behavior
The loss from progress bar during the training should be **mean** of loss across of all GPUs

### Environment

**Note**: `Bugs with code` are solved faster ! `Colab Notebook` should be made `public` !

* `IDE`: Please, use our python [bug_report_model.py](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report_model.py
) template.

* `Colab Notebook`: Please copy and paste the output from our [environment collection script](https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py) (or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py
# For security purposes, please check the contents of collect_env_details.py before running it.
python collect_env_details.py
```

 - PyTorch Version (e.g., 1.0):  1.4
 - OS (e.g., Linux):  Linux
 - How you installed PyTorch (`conda`, `pip`, source):  pip
 - Build command you used (if compiling from source):
 - Python version:  3.7
 - CUDA/cuDNN version:  10.1
 - GPU models and configuration:
 - Any other relevant information:   PyTorch Lightning 1.0.3

### LINKED ISSUE #9100: self.log in ddp model change 'reduce_fx' to 'mean',  'min' or 'max', but always got sum result
Dear Team:
As the title I said, the default behavior of self.log when sync_dist=True in ddp model is average all GPUs' results, but in the csv report I always got sum result.

wecom-temp-9dc4879d00d066eb655f11c4d84f16dd

wecom-temp-9237a589ba93491167f9bbdf105bf817

wecom-temp-01f033a582d2bd4005c5c7912396b9c7

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

See title.

Fixes #9100
Fixes #4395

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by ohayonguy** [comment]
I am still observing a difference between the logged loss and the loss returned by `training_step`:
image

Is this expected? `nt_xent_loss` is exactly the same as `loss`.

This is my training_step:
image

**Comment by tchaton** [comment]
Hey @ohayonguy, 

Lightning is adding a smoothing on the loss by computing a running mean over the logged loss within the progress bar.
Personally, I find it quite confusing. 
If you feel the same, please on an issue and ask for it to be removed :)

Best,
T.C

**Comment by carmocca** [comment]
> If you feel the same, please on an issue and ask for it to be removed :)

You can comment here: #9372
